LanguageFeatures/Static-access-shorthand/constant_expression_A10_t20.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A static member shorthand expression should be a potentially
+/// constant expression if the corresponding explicit static member plus
+/// selectors expression would be, which currently means that it's a potentially
+/// constant expression if and only if it's a constant expression.
+///
+/// @description Checks that a static member shorthand expression can be used in
+/// a constant `as` expression.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=enum-shorthands
+
+import '../../Utils/expect.dart';
+
+class C {
+  static const C one = const C(1);
+  final int v;
+  const C(this.v);
+}
+
+mixin M on C {
+  static const M one = const MC(1);
+}
+
+class MC extends C with M {
+  const MC(super.v);
+}
+
+enum E {
+  e1, e2;
+  static const E one = E.e1;
+}
+
+extension type const ET(int v) {
+  static const ET one = const ET(1);
+}
+
+main() {
+  const C c = .one as C;            // ignore: unnecessary_cast
```

Comment:
I think this will not work. The left hand operand of an `as` expression does not have a context type (other than `_`), and this means that `.one` is a compile-time error because we have no place to look up the name `one`.

Reply:
Deleted this test. I'll add `as` to operators tests with compile-time error as expected result.

Reply:
OK, sounds good!